packages/forms/resources/js/components/code-editor.js

```diff
@@ -5,15 +5,40 @@ import { html } from '@codemirror/lang-html'
 import { javascript } from '@codemirror/lang-javascript'
 import { json } from '@codemirror/lang-json'
 import { php } from '@codemirror/lang-php'
-import { EditorState } from '@codemirror/state'
+import { EditorState, Compartment } from '@codemirror/state'
 import { keymap } from '@codemirror/view'
+import { oneDark } from '@codemirror/theme-one-dark'
 
-export default function codeEditorFormComponent({ isDisabled, state }) {
+export default function codeEditorFormComponent({
+    isDisabled,
+    language,
+    state,
+}) {
     return {
         editor: null,
-
+        themeCompartment: new Compartment(),
         state,
 
+        isDarkMode: function () {
+            return document.documentElement.classList.contains('dark')
+        },
+
+        getThemeExtensions: function () {
+            return this.isDarkMode() ? [oneDark] : []
+        },
+
+        getLanguageExtension: function () {
+            const extensions = {
+                css,
+                html,
+                javascript,
+                json,
+                php,
+            }
+
+            return extensions[language]?.() || []
```

Comment:
Instead of returning an extension or an array, can we return an extension or null, and then when using the extension only register it if its not null?